packages/relay/src/lib/precheck.ts

```diff
@@ -376,4 +380,17 @@ export class Precheck {
       throw predefined.UNSUPPORTED_TRANSACTION_TYPE;
     }
   }
+
+  /**
+   * Checks if the receiver account exists.
+   * @param {Transaction} tx - The transaction.
+   * @param {RequestDetails} requestDetails - The request details for logging and tracking.
+   */
+  async receiverAccount(tx: Transaction, requestDetails: RequestDetails) {
+    const verifyAccount = await this.mirrorNodeClient.getAccount(tx.to!, requestDetails);
+
+    if (verifyAccount !== null && verifyAccount.receiver_sig_required === true) {
+      throw predefined.INTERNAL_ERROR("Receiver's signature is required.");
+    }
```

Comment:
Let's add another check to see if verifyAccount is null then throw a `RESOURCE_NOT_FOUND` and include the value of tx.to.

```
      throw predefined.RESOURCE_NOT_FOUND(`Cannot find receiver account: tx.to=${tx.to}`);
```

Reply:
Careful on this one since there are valid cases for `tx.to` not being set e.g. contract create.

Reply:
Ahh yeah you're right tx.to is nullable if the transaction is a contract create transaction. Then maybe in the test make the tx.to a non-null address but mock not found. @nadezhdapopovaa please let me know if this is clear enough

Reply:
Oh wait this block only executes when `tx.to` is not null; therefore, it is still possible for the mirror node to return a null result if `tx.to` is not found on the network. I'd suggest instead of checking `if (verifyAccount !== null && verifyAccount.receiver_sig_required === true)` for the sig required, we can pull the verifyAccount !== null out and add a new block to check the validity of the verifyAccount. If it's null then throw RESOURCE_NOT_FOUND like how we did with the tx.from.

Reply:
Please disregard the above! I completely overlooked the fact that `tx.to` can be a hollow address (with no associated account). We can send HBAR to that hollow account, and the system can lazily create the account as needed. Please proceed with your original flow. I apologize for any inconvenience this may have caused.